File: app/src/main/java/com/commit451/frankenstein/sample/LaunchActivity.java
package com.commit451.frankenstein.sample;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.widget.Toast;

public class LaunchActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Toast.makeText(this, "Launch activity launched", Toast.LENGTH_SHORT).show();
        startActivity(new Intent(this, MainActivity.class));
        finish();
    }
}

File: app/src/main/java/com/commit451/frankenstein/sample/LaunchActivity.kt
package com.commit451.frankenstein.sample

import android.content.Intent
import android.os.Build
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class LaunchActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        //Think of this being a long running operation to fetch the user, which we do on launch
        //and keep in static memory
        App.get().userName = Build.MODEL
        Toast.makeText(this, "Launch activity launched", Toast.LENGTH_SHORT)
                .show()
        startActivity(Intent(this, MainActivity::class.java))
        finish()
    }
}
